src/mon/FSCommands.cc

```diff
@@ -1525,6 +1513,33 @@ int FileSystemCommandHandler::_check_pool(
   }
 
   const string& pool_name = osd_map.get_pool_name(pool_id);
+  auto app_map = pool->application_metadata;
+
+  if(!allow_overlay && !app_map.empty()) {
+    auto app = app_map.find(pg_pool_t::APPLICATION_NAME_CEPHFS);
+    if(app != app_map.end()) {
+      auto& [app_name, app_metadata] = *app;
+      auto data = app_metadata.find("data");
+      auto metadata = app_metadata.find("metadata");
+      if(data != app_metadata.end()) {
+        auto& [pool_type, filesystem] = *data;
```

Comment:
Why using `pool_type` here (and below)? Using `type` makes more sense?

Reply:
Sure. Done.